subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -489,26 +490,37 @@ public TaskCreatingProvider(Class<T> type, String name, Action<? super T> config
         @Override
         public T getOrNull() {
             if (task == null) {
-                task = type.cast(findByNameWithoutRules(name));
-                if (task == null) {
-                    task = createTask(name, type, NO_ARGS);
-                    add(task);
-                    configureAction.execute(task);
-                    configureAction = null;
+                // We synchronize this to prevent multiple threads from attempting to create a task concurrently
+                synchronized (lock) {
```

Comment:
❌ This is not safe, double checked locking requires the field to be volatile. Same in the method below. Otherwise another thread could see a half-instantiated task.